prototypes/ScheduledMerges.hs

```diff
@@ -360,13 +364,36 @@ newLevelMerge tr level mergePolicy mergeLast rs = do
     -- merges will only complete at the last possible moment.
     -- Note that for levelling this is includes the single run in the current
     -- level.
-    debt = newMergeDebt $ case mergePolicy of
+    debt = case mergePolicy of
              MergePolicyLevelling -> 4 * tieringRunSize (level-1)
                                        + levellingRunSize level
              MergePolicyTiering   -> length rs * tieringRunSize (level-1)
     -- deliberately lazy:
     r    = mergek mergeType rs
 
+-- TODO: make more specific? this is only used inside the merging tree code
+-- and there should never be a MidLevel merge.
+newLeafMerge :: Tracer (ST s) EventDetail
+             -> MergeType -> [Run] -> ST s (MergingRun s)
+newLeafMerge tr mergeType rs = do
+    assertST (length rs > 1)
+    mrs <- case filter (\r -> runSize r > 0) rs of
+      []  -> return (CompletedMerge Map.empty)
+      [r] -> return (CompletedMerge r)
+      rs' -> do
+        let !cost = sum (map runSize rs')
+        let !debt = cost  -- TODO: is there a reason to claim a larger debt?
```

Comment:
So with normal runs we claim a debt that would cover the maximum cost, because we want to be sure we're supplying enough credits for the worst case.

But here, the cost is arbitrary, and the user has to supply enough credits to cover it over whatever time period they think makes sense. So yeah, I think debt = cost here is fine.